### ISSUE TITLE
Unexplained Crash

### ISSUE THREAD

**Issue by superjax** [byu-magicc/ROSflight#137]
We took the hardware out flying today, and we observed a random falling out of the sky.  When looking at the bag files, I can't see the fall, which makes me wonder if the firmware crashed.  It looked like the motors just stopped mid-flight.

Potential Causes:

My first guess is that the firmware crashed.  I don't really know how to replicate that on the ground though.  Perhaps HIL simulation?  (That seems like a lot of work for what is probably a simple bug).  I'll look for divide by zeros for a while, and potentially bad pointers, but I don't know what else could have caused a software crash.

We also lost WiFi to the multirotor, which means I can't know for sure why the bag recording stopped.  It could have stopped because it lost WiFi and terminated.

**Comment by superjax** [comment]
I ran the naze on my desk with an RC receiver and PWM for about 4 hours yesterday to try to see the crash.  I didn't see anything unusual.  It seemed to run just find, not large state jumps, no resetting, no freezing.  Granted, I am unable to perfectly replicate flight conditions on my desk, but the software didn't seem to freak out.

I also deliberately caused Hard Faults to happen.  They seem to just reset the flight controller, however PWM stops when a Hard Fault is entered. (Probably because the Hard Fault is highest priority).  I tried handling the Hard Fault (with a system reboot), but we might want to look into programming in a way to do a "hot restart" where the flight controller comes up armed if it restarted while armed.
